code/modules/mob/living/simple_animal/hostile/mining_mobs/wolf.dm

```diff
@@ -0,0 +1,60 @@
+/mob/living/simple_animal/hostile/asteroid/wolf
+	name = "white wolf"
+	desc = "A beast that survives by feasting on weaker opponents, they're much stronger with numbers."
+	icon = 'icons/mob/icemoon/icemoon_monsters.dmi'
+	icon_state = "whitewolf"
+	icon_living = "whitewolf"
+	icon_dead = "whitewolf_dead"
+	mob_biotypes = MOB_ORGANIC|MOB_BEAST
+	mouse_opacity = MOUSE_OPACITY_ICON
+	friendly_verb_continuous = "howls at"
+	friendly_verb_simple = "howl at"
+	speak_emote = list("howls")
+	speed = 5
+	move_to_delay = 5
+	maxHealth = 130
+	health = 130
+	obj_damage = 15
+	melee_damage_lower = 7.5
+	melee_damage_upper = 7.5
+	rapid_melee = 2 // every second attack
+	dodging = TRUE
+	dodge_prob = 50
+	attack_verb_continuous = "bites"
+	attack_verb_simple = "bite"
+	attack_sound = 'sound/weapons/bite.ogg'
+	vision_range = 7
+	aggro_vision_range = 7
+	move_force = MOVE_FORCE_WEAK
+	move_resist = MOVE_FORCE_WEAK
+	pull_force = MOVE_FORCE_WEAK
+	butcher_results = list(/obj/item/reagent_containers/food/snacks/meat/slab = 2, /obj/item/stack/sheet/sinew/wolf = 2, /obj/item/stack/sheet/bone = 2)
+	loot = list()
+	crusher_loot = /obj/item/crusher_trophy/watcher_wing
+	stat_attack = UNCONSCIOUS
+	robust_searching = TRUE
+	footstep_type = FOOTSTEP_MOB_CLAW
+	/// Message for when the wolf decides to start running away
+	var/retreat_message_said = FALSE
+
+/mob/living/simple_animal/hostile/asteroid/wolf/Move(atom/newloc)
+	if(newloc && newloc.z == z && (islava(newloc) || ischasm(newloc)))
+		return FALSE
+	return ..()
+
+/mob/living/simple_animal/hostile/asteroid/wolf/adjustHealth(amount, updating_health = TRUE, forced = FALSE)
+	. = ..()
+	if(health > maxHealth*0.1)
+		retreat_distance = initial(retreat_distance)
+		return
+	if(!retreat_message_said && target)
+		visible_message("<span class='danger'>The [name] tries to flee from [target]!</span>")
+		retreat_message_said = TRUE
+	retreat_distance = 30
```

Comment:
Couldn't this message happen right after the mob has died?

Reply:
its supposed to happen before the mob dies??

Reply:
`adjustHealth` calls the parent, which calls `updatehealth()` which calls `update_stat()` is that not correct?
Which means by the time of your message on the mob trying to flee it is possibly `stat == DEAD`